Unexpected behavior when more than two workers failed to run

### Description
Incorrect work manager, if some workers failed to run.

### Steps to reproduce

1. Use branch from PR https://github.com/bibendi/gruf-relay/pull/2
2. run test `go test -v --race ./internal/manager/...`

Expected behavior

```
➜  gruf-relay git:(manager-test-failed-workers) go test -v ./internal/manager/...
=== RUN   TestManager
Running Suite: Manager Suite - /other/gruf-relay/internal/manager
================================================================================
Random Seed: 1746630365

Will run 4 of 4 specs
•{"time":"2025-05-07T18:06:05.387969+03:00","level":"INFO","msg":"Starting manager","workers_count":2}
•{"time":"2025-05-07T18:06:06.390066+03:00","level":"INFO","msg":"Starting manager","workers_count":2}
{"time":"2025-05-07T18:06:06.390124+03:00","level":"ERROR","msg":"Failed to run worker","error":"failed to start process","worker":{}}
•{"time":"2025-05-07T18:06:06.390548+03:00","level":"INFO","msg":"Starting manager","workers_count":3}
{"time":"2025-05-07T18:06:06.390603+03:00","level":"ERROR","msg":"Failed to run worker","error":"failed to start process","worker":{}}
{"time":"2025-05-07T18:06:06.390614+03:00","level":"ERROR","msg":"Failed to run worker","error":"failed to start process","worker":{}}
•

Ran 4 of 4 Specs in 1.004 seconds
SUCCESS! -- 4 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestManager (1.01s)
PASS
ok  	github.com/bibendi/gruf-relay/internal/manager
```

Actual behavior

```
➜  gruf-relay git:(manager-test-failed-workers) go test -v --race ./internal/manager/...
=== RUN   TestManager
Running Suite: Manager Suite - /other/gruf-relay/internal/manager
================================================================================
Random Seed: 1746632033

Will run 4 of 4 specs
•{"time":"2025-05-07T18:33:53.442533+03:00","level":"INFO","msg":"Starting manager","workers_count":2}
•{"time":"2025-05-07T18:33:54.446492+03:00","level":"INFO","msg":"Starting manager","workers_count":2}
{"time":"2025-05-07T18:33:54.446731+03:00","level":"ERROR","msg":"Failed to run worker","error":"failed to start process","worker":{}}
•{"time":"2025-05-07T18:33:54.449892+03:00","level":"INFO","msg":"Starting manager","workers_count":3}
{"time":"2025-05-07T18:33:54.450066+03:00","level":"ERROR","msg":"Failed to run worker","error":"failed to start process","worker":{}}
==================
WARNING: DATA RACE
Read at 0x00c00027ef60 by goroutine 33:
  runtime.chansend1()
      ...

==================
panic: send on closed channel

goroutine 12 [running]:
github.com/bibendi/gruf-relay/internal/manager.(*Manager).Run.func1({0x100cba1b0, 0xc00000c5d0})
	/other/gruf-relay/internal/manager/manager.go:49 +0xd0
created by github.com/bibendi/gruf-relay/internal/manager.(*Manager).Run in goroutine 10
	/other/gruf-relay/internal/manager/manager.go:45 +0x2a4
FAIL	github.com/bibendi/gruf-relay/internal/manager	1.628s
FAIL
```

--- a/internal/manager/manager.go
+++ b/internal/manager/manager.go
@@ -32,7 +32,6 @@ func NewManager(cfg config.Workers) *Manager {
 
 func (m *Manager) Run(ctx context.Context) error {
 	var wg sync.WaitGroup
-	defer wg.Wait()
 
 	log.Info("Starting manager", slog.Int("workers_count", len(m.workers)))
 	errChan := make(chan error, 1)
@@ -61,6 +60,7 @@ func (m *Manager) Run(ctx context.Context) error {
 	}
 
 	cancel()
+	wg.Wait()
 	return err
 }
 

--- a/internal/manager/manager_test.go
+++ b/internal/manager/manager_test.go
@@ -6,11 +6,12 @@ import (
 	"testing"
 	"time"
 
-	"github.com/bibendi/gruf-relay/internal/config"
-	"github.com/bibendi/gruf-relay/internal/worker"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"go.uber.org/mock/gomock"
+
+	"github.com/bibendi/gruf-relay/internal/config"
+	"github.com/bibendi/gruf-relay/internal/worker"
 )
 
 func TestManager(t *testing.T) {
@@ -79,6 +80,35 @@ var _ = Describe("Manager", func() {
 			manager.workers = map[string]worker.Worker{
 				"worker-1": worker1,
 				"worker-2": worker2,
+			}
+
+			err := manager.Run(ctx)
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(Equal(expectedError))
+		})
+
+		It("returns an error if any workers fails to start", func() {
+			ctx := context.Background()
+			worker1 := worker.NewMockWorker(ctrl)
+			worker2 := worker.NewMockWorker(ctrl)
+			worker3 := worker.NewMockWorker(ctrl)
+
+			expectedError := errors.New("failed to start process")
+
+			worker1.EXPECT().Run(gomock.Any()).Return(expectedError)
+			worker1.EXPECT().String().Return("worker-1").AnyTimes()
+
+			worker2.EXPECT().Run(gomock.Any()).Return(nil).AnyTimes()
+			worker2.EXPECT().String().Return("worker-2").AnyTimes()
+
+			worker3.EXPECT().Run(gomock.Any()).Return(expectedError)
+			worker3.EXPECT().String().Return("worker-3").AnyTimes()
+
+			manager := NewManager(workersCfg)
+			manager.workers = map[string]worker.Worker{
+				"worker-1": worker1,
+				"worker-2": worker2,
+				"worker-3": worker3,
 			}
 
 			err := manager.Run(ctx)

